From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 3 Sep 2010 00:18:09 -0400
Subject: [net] vxge: use pci_dma_mapping_error to test return val
Message-id: <20100903001808.16045.36245.stgit@localhost6.localdomain6>
Patchwork-id: 28045
O-Subject: [RHEL5.6 BZ608598 PATCH 04/15] vxge: use pci_dma_mapping_error to
	test return value
Bugzilla: 608598
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Bob Picco <bpicco@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

From: Denis Kirjanov <kirjanov@gmail.com>

pci_dma_mapping_error should be used to test return value of
pci_map_single or pci_map_page.

Signed-off-by: Denis Kirjanov <kirjanov@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit fa15e99b6bb44aa86b241a43ca8c509e91f80153)

diff --git a/drivers/net/vxge/vxge-main.c b/drivers/net/vxge/vxge-main.c
index 22bdc81..cdc6b23 100644
--- a/drivers/net/vxge/vxge-main.c
+++ b/drivers/net/vxge/vxge-main.c
@@ -306,7 +306,7 @@ static int vxge_rx_map(void *dtrh, struct vxge_ring *ring)
 	dma_addr = pci_map_single(ring->pdev, rx_priv->skb_data,
 				rx_priv->data_size, PCI_DMA_FROMDEVICE);
 
-	if (dma_addr == 0) {
+	if (unlikely(pci_dma_mapping_error(dma_addr))) {
 		ring->stats.pci_map_fail++;
 		return -EIO;
 	}
